presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformUncorrelatedInPredicateSubqueryToSemiJoin.java

```diff
@@ -106,7 +107,8 @@ public PlanNode visitApply(ApplyNode node, RewriteContext<Void> context)
                     semiJoinSymbol,
                     Optional.empty(),
                     Optional.empty(),
-                    Optional.empty()
+                    Optional.empty(),
+                    SystemSessionProperties.isLegacySemiJoin(context.get())
```

Comment:
The idea is that you can think of `SemiJoinNode` with the flag on and off as two kind of `SemiJoinNodes`.
For instance, `ApplyNode` transformation would set the `SemiJoinNode` flag accordingly (e.g: depending if the semi join output is used in projection or filter) and other optimizers should obey the flag (e.g: predicate pushdown). This way you decouple symbolic analysis from other predicate pushdown. This is one of the ways forward if we want to further enhance lhs support null in semi join.

I don't have strong option on it though. You can remove the flag from `SemiJoinNode` altogether